pkg/event/internal/querytools/apply_query_field.go

```diff
@@ -13,13 +13,25 @@ import (
 )
 
 func ApplyQueryField(fieldPtr interface{}, queryField event.QueryField) error {
-	if reflecttools.IsZero(queryField) {
-		return event.ErrQueryFieldHasZeroValue{QueryField: queryField}
-	}
 	field := reflect.ValueOf(fieldPtr).Elem()
```

Comment:
I'm no `reflect` expert, but I tried passing an empty string pointer to this and it `panic`'ed. Unless I'm missing something, is this any risk we are subject to? 

Reproduce at https://play.golang.org/p/7-8KZbbxFbG

Reply:
> Reproduce at https://play.golang.org/p/7-8KZbbxFbG

Specifically in this sandbox it panics because of `Elem()` on non-pointer (and if you will replace `X.Elem()` with `reflect.Indirect(X)` then it won't panic).

In our case here:
```go
field := reflect.ValueOf(fieldPtr).Elem()
```
`fieldPtr` is always a pointer.

Reply:
thanks for the explanation, just one more question: `fieldPtr` is of type `interface{}`, which is not guaranteed to be of type `Ptr` , is that correct? And if so this could still panic?

Reply:
> which is not guaranteed to be of type `Ptr`, is that correct?

It is guaranteed through unit-tests. Yes: it is not as reliable as compile-time check. It will be possible to introduce compile-time checks in Go2.

If you want I can unroll this generic solution to a multiple particular solutions (to eliminate dynamic typing).

Reply:
no, that's good enough, thanks for the explanation